connectivity/factory/factory.go

```diff
@@ -0,0 +1,248 @@
+	clientEgressToCIDRExternalDenyPolicyYAML string
+
+	//go:embed manifests/client-egress-l7-http.yaml
+	clientEgressL7HTTPPolicyYAML string
+
+	//go:embed manifests/client-egress-l7-http-named-port.yaml
+	clientEgressL7HTTPNamedPortPolicyYAML string
+
+	//go:embed manifests/client-egress-l7-tls.yaml
+	clientEgressL7TLSPolicyYAML string
+
+	//go:embed manifests/client-egress-l7-http-matchheader-secret.yaml
+	clientEgressL7HTTPMatchheaderSecretYAML string
+
+	//go:embed manifests/echo-ingress-from-cidr.yaml
+	echoIngressFromCIDRYAML string
+)
+
+var (
+	clientLabel  = map[string]string{"name": "client"}
+	client2Label = map[string]string{"name": "client2"}
+
+	// Test order is important
+	primaryTests = []factory{
+		noUnexpectedPacketDrops{},
+		noPolicies{},
+		noPoliciesFromOutside{},
+		noPoliciesExtra{},
+		allowAllExceptWorld{},
+		clientIngress{},
+		clientIngressKnp{},
+		allowAllWithMetricsCheck{},
+		allIngressDeny{},
+		allIngressDenyFromOutside{},
+		allIngressDenyKnp{},
+		allEgressDeny{},
+		allEgressDenyKnp{},
+		allEntitiesDeny{},
+		clusterEntity{},
+		clusterEntityMultiCluster{},
+		hostEntityEgress{},
+		hostEntityIngress{},
+		echoIngress{},
+		echoIngressFromOutside{},
+		echoIngressKnp{},
+		clientIngressIcmp{},
+		clientEgress{},
+		clientEgressKnp{},
+		clientEgressExpression{},
+		clientEgressExpressionKnp{},
+		clientWithServiceAccountEgressToEcho{},
+		clientEgressToEchoServiceAccount{},
+		toEntitiesWorld{},
+		toCidrExternal{},
+		toCidrExternalKnp{},
+		fromCidrHostNetns{},
+		echoIngressFromOtherClientDeny{},
+		clientIngressFromOtherClientIcmpDeny{},
+		clientEgressToEchoDeny{},
+		clientIngressToEchoNamedPortDeny{},
+		clientEgressToEchoExpressionDeny{},
+		clientWithServiceAccountEgressToEchoDeny{},
+		clientEgressToEchoServiceAccountDeny{},
+		clientEgressToCidrDeny{},
+		clientEgressToCidrDenyDefault{},
+		health{},
+		northSouthLoadbalancing{},
+		podToPodEncryption{},
+		nodeToNodeEncryption{},
+		egressGateway{},
+		egressGatewayExcludedCidrs{},
+		podToNodeCidrpolicy{},
+		northSouthLoadbalancingWithL7Policy{},
+		echoIngressL7{},
+		echoIngressL7NamedPort{},
+		clientEgressL7Method{},
+		clientEgressL7{},
+		clientEgressL7NamedPort{},
+		clientEgressL7TlsDenyWithoutHeaders{},
+		clientEgressL7TlsHeaders{},
+		clientEgressL7SetHeader{},
+		echoIngressAuthAlwaysFail{},
+		echoIngressMutualAuthSpiffe{},
+		podToIngressService{},
+		podToIngressServiceDenyAll{},
+		podToIngressServiceDenyIngressIdentity{},
+		podToIngressServiceDenyBackendService{},
+		podToIngressServiceAllowIngressIdentity{},
+		outsideToIngressService{},
+		outsideToIngressServiceDenyWorldIdentity{},
+		outsideToIngressServiceDenyCidr{},
+		outsideToIngressServiceDenyAllIngress{},
+		dnsOnly{},
+		toFqdns{},
+		podToControlplaneHost{},
+		podToK8sOnControlplane{},
+		podToControlplaneHostCidr{},
+		podToK8sOnControlplaneCidr{},
+	}
+)
+
+type factory interface {
+	name() string
+	shouldRun(version semver.Version, params check.Parameters) bool
```

Comment:
`Test` struct already has `willRun` https://github.com/cilium/cilium-cli/blob/main/connectivity/check/test.go#L234 that checks for cilium version and such. it will be confusing to have both this method and willRUn.

Reply:
@michi-covalent we are planning to expand on this a bit more. We had this session from @fgiloux during the hive time, where we discussed the possibilities to group the tests based on labeling them. Imagine have a label "single-node", or "clustermes" or whatever. Today this filtering is done based on naming assumptions, we want to change that and allow for more finer grained bisecting the whole test catalog.

Reply:
Also, I don't want to do a logic refactoring in this PR. It's already big enough, isn't it?

Reply:
then i'd suggest doing necessary refactoring first in a separate PR. i think it's more natural for this `shouldRun` method to belong to `Test` instead of `TestBuilder` 💭

Reply:
`factory` interface renamed to `testBuilder` and `factory` package renamed to `builder`.

Reply:
I've removed `shouldRun` method and added `WithConditionFn` method to the Test struct.

Reply:
i love it